module/mailnews/newsread.c

```
@@ -389,8 +412,132 @@ Bool DateFromSeconds(long seconds, char *str)
+
+   switch (nColumn)
+   {
+   case COL_TITLE:
+      return bSortAscending ? stricmp(title1, title2) : stricmp(title2, title1);
+   case COL_POSTER:
+      return bSortAscending ? stricmp(poster1, poster2) : stricmp(poster2, poster1);
+   case COL_TIME:
+      return bSortAscending ? article1->time - article2->time : article2->time - article1->time;
+   }
+
+   return 0;
+}
+
+/****************************************************************************/
+/*
+ * UpdateColumnHeaders:  Given a list view handle, the index of the sorted 
+ *   column, and a boolean indicating the sort direction, update the header 
+ *   of the list view columns to indicate the sorted column.
+ *
+ * Parameters:
+ *   hListView - Handle to the list view control.
+ *   sortedColumn - Index of the column that is sorted.
+ *   sortAscending - Boolean indicating if the sort direction is ascending 
+ *     (TRUE) or descending (FALSE).
+ */
+void UpdateColumnHeaders(HWND hListView, int sortedColumn, BOOL sortAscending)
+{
+   // Get the handle to the header control associated with the list view
+   HWND hHeader = ListView_GetHeader(hListView);
+   if (!hHeader)
+       return;  // Ensure we have a valid header handle
+
+   // Get the number of columns in the header
+   int columnCount = Header_GetItemCount(hHeader);
+
+   for (int i = 0; i < columnCount; i++)
+   {
+      HDITEM hdi = {0};
+      hdi.mask = HDI_FORMAT | HDI_BITMAP;
+      
+      // Get the current format of the header item
+      Header_GetItem(hHeader, i, &hdi);
+
+      if (i == sortedColumn)
+      {
+         UINT bitmapID = sortAscending ? IDB_UPARROW : IDB_DOWNARROW;
+
+         hdi.fmt |= HDF_BITMAP | HDF_BITMAP_ON_RIGHT;
+
+         HBITMAP hBitmap = (HBITMAP)LoadImage(GetModuleHandle(NULL), MAKEINTRESOURCE(bitmapID), IMAGE_BITMAP, 0, 0, LR_LOADTRANSPARENT | LR_LOADMAP3DCOLORS);
+         if (hBitmap == NULL)
+         {
+            MessageBox(NULL, "Failed to load bitmap!", "Error", MB_OK | MB_ICONERROR);
```

This error message is kind of a "PC LOAD LETTER".  What is the user supposed to do?

Just silently fail and don't draw a bitmap in this case.